pkg/sqlite/configmap_writer.go

```diff
@@ -0,0 +1,212 @@
+package sqlite
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+
+	"github.com/ghodss/yaml"
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/sirupsen/logrus"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/operator-framework/operator-registry/pkg/client"
+)
+
+// configmap keys can contain underscores, but configmap names can not
+var unallowedKeyChars = regexp.MustCompile("[^-A-Za-z0-9_.]")
+
+const (
+	ResourceKey                 = "operators.coreos.com.bundle.resources"
+	MediaTypeKey                = "operators.coreos.com.bundle.mediatype"
+	EnvContainerImage           = "CONTAINER_IMAGE"
+	ConfigMapImageAnnotationKey = "olm.sourceImage"
+)
+
+type AnnotationsFile struct {
+	Annotations struct {
+		Resources string `json:"operators.coreos.com.bundle.resources"`
+		MediaType string `json:"operators.coreos.com.bundle.mediatype"`
+	} `json:"annotations"`
+}
+
+type ConfigMapWriter struct {
+	manifestsDir  string
+	configMapName string
+	namespace     string
+	clientset     *kubernetes.Clientset
+}
+
+var _ SQLPopulator = &ConfigMapWriter{}
+
+func NewConfigMapLoaderForDirectory(configMapName, namespace, manifestsDir, kubeconfig string) *ConfigMapWriter {
+	clientset, err := client.NewKubeClient(kubeconfig, logrus.StandardLogger())
+	if err != nil {
+		logrus.Fatalf("cluster config failed: %v", err)
+	}
+
+	return &ConfigMapWriter{
+		manifestsDir:  manifestsDir,
+		configMapName: configMapName,
+		namespace:     namespace,
+		clientset:     clientset,
+	}
+}
+
+func TranslateInvalidChars(input string) string {
+	validConfigMapKey := unallowedKeyChars.ReplaceAllString(input, "~")
```

Comment:
It is not apparent to me why you are providing `~` as an input (even after googling). Could you explain?

Reply:
This ties into the above comment. If you want to argue that failing instead of translating to valid is best, then this code is up for potential removal. But I randomly chose the tilda character as a character that is typically rarely used to replace any invalid characters. In practice, I hope the string is not modified, but did make sure to include a warning message to point out the potential issue.